### PR TITLE
Added `gabbard.py` in `poliastro.plotting`

### PR DESCRIPTION
**Pull request by TheBuffer**
Tried adding a Static Gabbard Plotter as described in #1431
and tested it out with the COSMOS 1408 DEB data
Screenshot from 2022-02-04 14-12-07

### CODE REVIEW

**Review by astrojuanlu — COMMENTED** [review]
Thanks @TheBuffer for this contribution! 🎉 I rebased your branch and everything looks good initially. I'm going to review this in depth in a few hours.

**Comment by TheBuffer** [comment]
Should I try adding the Dynamic Gabbard Plots and maybe integrating something which takes in the Debris Ephemerides as an input?

File: src/poliastro/plotting/gabbard.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        if not self._ax:
+            if dark:
+                with plt.style.context("dark_background"):
+                    _, self._ax = plt.subplots(figsize=(6, 6))
+            else:
+                _, self._ax = plt.subplots(figsize=(6, 6))
+
+        self._frame = None
+
+    @staticmethod
```
Comment (astrojuanlu) [inline src/poliastro/plotting/gabbard.py:20]: I'd say we don't need this static method? (See my comment below)

File: src/poliastro/plotting/gabbard.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    def _get_orbit_properties(orbit):
+        return orbit.r_p, orbit.r_a, orbit.period
+
+    def _get_orbit_property_list(self, orbits):
+        apogees = []
+        perigees = []
+        periods = []
+
+        for orbit in orbits:
+            perigee, apogee, period = self._get_orbit_properties(orbit)
```
Comment (astrojuanlu) [inline src/poliastro/plotting/gabbard.py:30]: We could simplify this to remove the static method:

```suggestion
            perigee, apogee, period = orbit.r_p, orbit.r_a, orbit.period
```

File: src/poliastro/plotting/gabbard.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        apogee_paths = plt.scatter(
+            periods, apogees, marker="o", color="blue", label="Apogee"
+        )
+        perigee_paths = plt.scatter(
+            periods, perigees, marker="o", color="red", label="Perigee"
+        )
+
+        self._ax.set_xlabel("Period(min)")
+        self._ax.set_ylabel("Altitude(km)")
```
Comment (astrojuanlu) [inline src/poliastro/plotting/gabbard.py:53]:
```suggestion
        self._ax.set_xlabel("Period (min)")
        self._ax.set_ylabel("Altitude (km)")
```

(see comment about units above)

File: src/poliastro/plotting/gabbard.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    def _get_orbit_property_list(self, orbits):
+        apogees = []
+        perigees = []
+        periods = []
+
+        for orbit in orbits:
+            perigee, apogee, period = self._get_orbit_properties(orbit)
+            apogees.append(apogee.value)
+            perigees.append(perigee.value)
+            periods.append(period.to(u.min).value)
```
Comment (astrojuanlu) [inline src/poliastro/plotting/gabbard.py:33]: Probably we should not hardcode `u.min` here and make it a parameter, like we did for other plotters in #1318

File: src/poliastro/plotting/gabbard.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        return apogees, perigees, periods
+
+    def _static_gabbard_plot(self, orbits):
+        """Plots a Static Gabbard Plot given a list of Orbits
+        Parameters
+        ----------
+        orbits : ~poliastro.twobody.orbit.Orbit List
+            The Orbits whose perigee and apogee will be plotted.
+        """
```
Comment (astrojuanlu) [inline src/poliastro/plotting/gabbard.py:42]:
```suggestion
        """Plots a Static Gabbard Plot given a list of Orbits

        Parameters
        ----------
        orbits : ~poliastro.twobody.orbit.Orbit List
            The Orbits whose perigee and apogee will be plotted.

        """
```

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
This looks very good, thanks @TheBuffer for your contribution! I'd say let's focus on static Gabbard plots for now to make things simpler, we can expand it later.

I left some comments, but most importantly this will need some tests. You can take inspiration from the image tests we have for other plotters, that compare an expected outcome with the actual one:

https://github.com/poliastro/poliastro/tree/main/tests/tests_plotting

File: src/poliastro/examples.py [hunk 6]
```diff
@@ -58,3 +58,186 @@
     epoch=time.Time("2015-11-05 12:00", scale="utc"),
 )
 """Comet 67P/Churyumov–Gerasimenko orbit example"""
+
+debris_orbits_list = []
```
Comment (astrojuanlu) [inline src/poliastro/examples.py:62]: I'd prefer that we put this only in the tests, rather than shipping it with the library.

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Starting to look good! Left only one comment. With that, I'm 👍🏽 as soon as the tests pass.

**Comment by astrojuanlu** [comment]
There seem to be some subtle (unimportant) typography differences between the baseline image and the result. I downloaded `result.png` via SSH, it's this:

result

and the difference is this:

result-failed-diff

You can probably just download `result.png` and commit it as the baseline.

**Review by astrojuanlu — APPROVED** [review]
Looks good to me! Thanks again @TheBuffer !

**Pull request merged**